core/spec/models/spree/price_spec.rb

```diff
@@ -1,6 +1,63 @@
 require 'spec_helper'
 
 describe Spree::Price, type: :model do
+  it { is_expected.to respond_to(:valid_from) }
+
+  describe 'scopes' do
+    describe '.valid_before(date)' do
+      let!(:past_price) { create(:price, valid_from: "1970-01-01") }
+      let!(:current_price) { create(:price, valid_from: "1990-01-01") }
+      let!(:future_price) { create(:price, valid_from: "2070-01-01") }
+
+      subject { described_class.valid_before("2012-01-01") }
+
+      it { is_expected.to contain_exactly(past_price, current_price) }
+    end
+
+    describe '.default' do
+      let!(:past_price) { create(:price, valid_from: "1970-01-01") }
+      let!(:current_price) { create(:price, valid_from: "1990-01-01") }
+      let!(:future_price) { create(:price, valid_from: "2070-01-01") }
+
+      before do
+        expect(Time).to receive(:current).and_return("2012-01-01")
+      end
+
+      subject { described_class.default }
+
+      it 'takes the last price that is not in the future' do
+        expect(subject).to eq(current_price)
+      end
+    end
+  end
+
+  describe 'intialization' do
+    let(:variant) { create(:variant) }
+    subject { described_class.new(variant: variant, valid_from: valid_from) }
+
+    context 'with a valid from date given' do
+      let(:valid_from) { "1990-01-01" }
+
+      it 'stays the same' do
+        subject.save
+        expect(subject.valid_from).to eq(Date.new(1990, 01, 01))
+      end
+    end
+
+    context 'with no valid from date given' do
+      let(:valid_from) { nil }
+
+      before do
+        expect(Time).to receive(:current).at_least(:once).and_return(Date.new(2015, 02, 28))
```

Comment:
I would prefer to use the Timecop library for this, which is already a dependency in Solidus.